stravalib/model.py

```diff
@@ -305,64 +308,57 @@ class AthleteStats(
         "biggest_climb_elevation_gain": "meters",
     }
 
+    @validator(
+        "recent_ride_totals",
+        "recent_run_totals",
+        "recent_swim_totals",
+        "ytd_ride_totals",
+        "ytd_run_totals",
+        "ytd_swim_totals",
+        "all_ride_totals",
+        "all_run_totals",
+        "all_swim_totals",
+    )
+    def to_extended_totals(cls, raw_totals: Dict):
+        return ActivityTotals.parse_obj(raw_totals)
```

Comment:
Using pydantic's validator mechanism, we can intercept and modify the parsing of fields. In this case, we can make sure that nested entities will be of our backward-compatible extended types instead of the "plain" generated pydantic model type.

Reply:
@jsamoocha i'm working through all of these elements but when we grab athlete stats via the api call is it possible that it only grabs public activities? 

my numbers are still incorrect and i suspect it's because i don't always make things public. is that how ytd_ ... totals are supposed to pull data in? I looked at my scope and `read_all` should include public and private activities as i understand it.

So far everything is working well but the data i'm getting is just not right. i'm going to look at specific activities next.

Reply:
I just tested and can confirm that Strava only counts our public activities in these ytd_* totals. I'll add that to the documentation.

Reply:
ok perfect. thanks.